From 1a5bdaa84a306c1eedf22b5b7879914a76beb803 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Sat, 22 Sep 2018 18:46:07 +0200
Subject: [PATCH 2/3] tests: support older python versions

assertRaisesRegexp was renamed to assertRaisesRegex in python 3.2.
This patch tries to determine the python version and call appropriate
assertRaiseRegex* method
---
 tests/test-umockdev.py | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/tests/test-umockdev.py b/tests/test-umockdev.py
index 2cadaad..58d8bab 100644
--- a/tests/test-umockdev.py
+++ b/tests/test-umockdev.py
@@ -190,12 +190,19 @@ A: simple_attr=1
             self.assertEqual(f.read(), b'\x41\xFF\x00\x05\xFF\x00')
 
     def test_add_from_string_errors(self):
+        try:
+            # python 3.2+
+            assertRaisesRegex = self.assertRaisesRegex
+        except AttributeError:
+            # python 2 && python < 3.2
+            assertRaisesRegex = self.assertRaisesRegexp
+
         # does not start with P:
-        with self.assertRaisesRegex(GLib.GError, 'must start with.*P:') as cm:
+        with assertRaisesRegex(GLib.GError, 'must start with.*P:') as cm:
             self.testbed.add_from_string ('E: SIMPLE_PROP=1\n')
 
         # no value
-        with self.assertRaisesRegex(GLib.GError, 'malformed attribute') as cm:
+        with assertRaisesRegex(GLib.GError, 'malformed attribute') as cm:
             self.testbed.add_from_string ('P: /devices/dev1\nE: SIMPLE_PROP\n')
 
 unittest.main(testRunner=unittest.TextTestRunner(stream=sys.stdout, verbosity=2))
-- 
2.17.1

